satpy/readers/file_handlers.py

```diff
@@ -20,9 +20,30 @@
 from abc import ABCMeta
 
 import numpy as np
+import xarray as xr
 from pyresample.geometry import SwathDefinition
 
 from satpy.dataset import combine_metadata
+from satpy.readers import open_file_or_filename
+
+
+def open_dataset(filename, *args, **kwargs):
+    """Open a file with xarray.
+
+    Args:
+       filename (Union[str, FSFile]):
+           The path to the file to open. Can be a `string` or
+           :class:`~satpy.readers.FSFile` object which allows using
+           `fsspec` or `s3fs` like files.
+
+    Returns:
+       xarray.Dataset:
+
+    Notes:
+       This can be used to enable readers to open remote files.
+    """
+    f_obj = open_file_or_filename(filename)
+    return xr.open_dataset(f_obj, *args, **kwargs)
```

Comment:
Could you add a small test for this?

Reply:
I think this refactor probably is the reason why some tests in the seviri nc reader are failing now too. I will have a look at it and will try to fix it.

Reply:
@BENR0 did you have some time to look at this?

Reply:
Sorry I didn't have time to look at it yet due to a project deadline. Maybe I have a chance later today otherwise I will look at it during the weekend.

Reply:
no problem, take it easy :)